W-16787687-supported-bindings-apikit-asyncapi-CP (#205)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-for-asyncapi-supported-bindings.adoc

= Supported Bindings and Parametrized Channels in APIkit for AsyncAPI 
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

APIkit for AsyncAPI supports various bindings to enable message-driven architectures.

[%header%autowidthspread]
|===

| Connector | Publish | Message Listener | Description | Reference

| AnypointMQ | `destination` | `destination` | 

> The AnypointMQ connector defaults to using the channel name as the `destination`. The `destination` (queue or exchange) name for this channel must be specified if the channel name differs from the actual `destination` name or if the channel name isn’t a valid `destination` name.

Reviewer: ```suggestion
By default, the AnypointMQ connector uses the channel name as the `destination`. You must specify the `destination` (queue or exchange) name if the channel name differs from the actual `destination` name, or if the channel name isn’t a valid `destination` name.
```

Author: Great rewording! Thanks!

| See https://github.com/asyncapi/bindings/blob/b1e7736071ccddc6c3d652bb2071cad4549f8a08/anypointmq/README.md[Anypoint MQ Bindings].

.3+| Solace

| `deliveryMode` | `deliveryMode` | The Solace connector operates in `direct` or `persistent` mode. The default mode is `persistent`. The `deliveryMode` parameter determines the quality of service for publishing messages. The `Direct` mode offers lower latency but doesn’t guarantee message delivery. The ‘persistent’ mode ensures that messages are reliably delivered, even in the event of a broker failure.

| See https://github.com/asyncapi/bindings/blob/master/solace/README.md[Solace Bindings].

| `destinationType` | `destinationType` | The Solace connector supports `queue` in APIkit for AsyncAPI. | See https://github.com/asyncapi/bindings/blob/master/solace/README.md[Solace Bindings].

> | N/A | `topicSubscriptions` | The Solace connector specifies the list of topics to which the queue subscribes if the `destinationType` is `queue`. If you specify no topic, the `queue` automatically subscribes to the topic the channel name represents. | See https://github.com/asyncapi/bindings/blob/master/solace/README.md[Solace Bindings].

Reviewer: ```suggestion
| N/A | `topicSubscriptions` | The Solace connector specifies the list of topics the queue subscribes to when the `destinationType` is `queue`. If you specify no topic, the `queue` automatically subscribes to the topic the channel name represents. | See https://github.com/asyncapi/bindings/blob/master/solace/README.md[Solace Bindings].
```

Author: OK! Thanks!
